tests/test_util.py

```diff
@@ -456,3 +460,49 @@ def test_df_parser(tmpdir):
     assert len(df.labels) == 1
     assert df.labels.get('label') == 'foobar barfoo'
 
+def test_df_parser_parent_env_arg(tmpdir):
+    tmpdir_path = str(tmpdir.realpath())
+    p_env = {
+        "test_env": "first"
+    }
+    df_content = dedent("""\
+        FROM fedora
+        ENV foo=bar
+        LABEL label="foobar $test_env"
+        """)
+
+    with open(tmpdir_path + '/Dockerfile', 'w') as dockerfile:
+        dockerfile.write(df_content)
+        dockerfile.flush()
+
+    df = df_parser(tmpdir_path, parent_env=p_env)
+    assert df.labels.get('label') == 'foobar first'
+
+@pytest.mark.parametrize('env_arg', [
+    {"test_env": "first"},
+    ['test_env=first'],
```

Comment:
See the actual inspect output: it has to be a list of dicts (not a dict like in a first option and not a list of strings like in the last option). 

Instead it should test that 
* an empty list is parsed correctly (no envs in the base image), 
* a list of strings separated by '='